### PR TITLE
add: two sum

### PR DESCRIPTION
**Pull request by ongzx**
This MR adds a new leetcode Easy question (two sum: https://leetcode.com/problems/two-sum/) to the repo.

### CODE REVIEW

File: LeetcodeProblems/Algorithms/2Sum.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * @param {number[]} nums
+ * @param {number} target
+ * @return {number[]}
+ */
+ var twoSum = function(nums, target) {
+  let map ={};
+  for(let i=0;i<nums.length;i++){
+      const sum = target-nums[i];
+      if(map[parseInt(sum)] != void 0){
```
Comment (ignacio-chiazzo) [inline LeetcodeProblems/Algorithms/2Sum.js:35]: do we really need to use `void  0` or just `0`?

File: README.md [hunk 2]
```diff
@@ -76,6 +76,7 @@ To run a specific problem in your console run  `node <problem_file_path>` (e.g.
 | [Binary Gap ](/LeetcodeProblems/Algorithms/Binary_Gap.js)                                                       | Easy   | https://leetcode.com/problems/binary-gap/ |
 | [Binary Gap ](/LeetcodeProblems/Algorithms/Binary_Gap.js)                                                       | Easy   | https://leetcode.com/problems/binary-gap/ |
 | [Majority Element](/LeetcodeProblems/Algorithms/Majority_Element.js)                                            | Easy   | https://leetcode.com/problems/majority-element/ |
+| [Two Sum](/LeetcodeProblems/Algorithms/2Sum.js)                                            | Easy   | https://leetcode.com/problems/two-sum/ |
```
Comment (ignacio-chiazzo) [inline README.md:79]: Could you ident this please?

**Review by ignacio-chiazzo — COMMENTED** [review]
Hey, thanks for contributing. The solution looks good to me.  If we want to save memory in favour of computation we could sort the numbers and use two pointers technique.

Could you add a test file? https://github.com/ignacio-chiazzo/Algorithms-Leetcode-Javascript/issues/49#issuecomment-1265792953

File: LeetcodeProblems/Algorithms/2Sum.js (hunk 1 above)
Comment (ongzx) [reply LeetcodeProblems/Algorithms/2Sum.js]: 0 is fine too

File: README.md (hunk 2 above)
Comment (ongzx) [reply README.md]: sure

**Review by ignacio-chiazzo — APPROVED** [review]

**Pull request merged**